meta-lxatac-bsp: barebox: update to 2023.10.0

hnez opened the pull request:
This barebox version imports our mainline LXA TAC devicetrees from linux, greatly reducing our local patch stack size. It also contains some patches that have since been mainlined, also reducing our patch stack size.

TODO before merging:

- [x] Wait for barebox 2023.10.0 to be released …
- [x] … and reabase the patch stack on top of it
- [x] … and change the SRC_URIs to the released tarball
- [x] Test on actual hardware

hnez committed:
meta-lxatac-bsp: barebox: update to 2023.10.0

This barebox version imports our mainline LXA TAC devicetrees from
linux, greatly reducing our local patch stack size.
It also contains some patches that have since been mainlined,
also reducing our patch stack size.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez commented:
Based on the console output I would say that the updated barebox looks healthy:

```
NOTICE:  CPU: STM32MP157C?? Rev.Z
NOTICE:  Model: Linux Automation Test Automation Controller (TAC)
WARNING: VDD unknown
INFO:    Reset reason (0x15):
INFO:      Power-on Reset (rst_por)
INFO:    FCONF: Reading TB_FW firmware configuration file from: 0x2ffe2000
INFO:    FCONF: Reading firmware configuration information for: stm32mp_io
INFO:    Using EMMC
INFO:      Instance 2
INFO:    Boot used partition fsbl1
NOTICE:  BL2: v2.7(release):
NOTICE:  BL2: Built : 15:02:46, Jun  1 2022
INFO:    BL2: Doing platform setup
INFO:    RAM: DDR3-DDR3L 16bits 533000kHz
INFO:    Memory size = 0x20000000 (512 MB)
INFO:    BL2: Loading image id 31
INFO:    Loading image id=31 at address 0x2ffff000
INFO:    Image id=31 loaded: 0x2ffff000 - 0x2ffff226
INFO:    FCONF: Reading FW_CONFIG firmware configuration file from: 0x2ffff000
INFO:    FCONF: Reading firmware configuration information for: dyn_cfg
INFO:    FCONF: Reading firmware configuration information for: stm32mp1_firewall
INFO:    BL2: Loading image id 4
INFO:    Loading image id=4 at address 0x2ffc5000
INFO:    Image id=4 loaded: 0x2ffc5000 - 0x2ffcdb10
INFO:    BL2: Skip loading image id 21
INFO:    BL2: Skip loading image id 22
INFO:    BL2: Loading image id 23
INFO:    Loading image id=23 at address 0xc0500000
INFO:    Image id=23 loaded: 0xc0500000 - 0xc0500226
INFO:    BL2: Loading image id 26
INFO:    Loading image id=26 at address 0x2ffc0000
INFO:    Image id=26 loaded: 0x2ffc0000 - 0x2ffc3eaf
INFO:    BL2: Loading image id 5
INFO:    Loading image id=5 at address 0xc0100000
INFO:    Image id=5 loaded: 0xc0100000 - 0xc018af4b
NOTICE:  BL2: Booting BL32
INFO:    Entry point address = 0x2ffc5000
INFO:    SPSR = 0x1d3
NOTICE:  SP_MIN: v2.7(release):
NOTICE:  SP_MIN: Built : 15:02:46, Jun  1 2022
INFO:    ARM GICv2 driver initialized
INFO:    SP_MIN: Initializing runtime services
INFO:    SP_MIN: Preparing exit to normal world


barebox 2023.09.0-20231009-1-00105-g7f147d2ef6ed-dirty #1 Mon Oct 9 10:27:20 UTC 2023
Buildsystem version: 6b6cbb5-dirty

Board: Linux Automation Test Automation Controller (TAC) Gen 2
deep-probe: supported due to lxa,stm32mp157c-tac-gen2
stm32mp-init: detected STM32MP157C?? Rev.Z
STM32 RCC reset reason POR (MP_RSTSR: 0x00000015)
netconsole: registered as netconsole-1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 2
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
persistent_ram: error in header, 1
pstore: Registered ramoops as persistent store backend
ramoops: attached 0x200000@0xdfdd0000, ecc: 16/0
psci psci.of: detected version 1.1
usb251xb usb2514b0: Hub configuration was successful.
usb251xb usb2514b0: Hub probed successfully
stpmic1-i2c stpmic10: PMIC Chip Version: 0x21
v3v3_hdmi: Bringing 1800000uV into 3300000-3300000uV
v1v2_hdmi: Bringing 1000000uV into 1200000-1200000uV
registered as fbconsole0
ksz9477-spi switch@00: Found KSZ9893 or compatible
ksz9477-spi switch@00: chip id: 0x00989360
mdio_bus: miibus0: probed
mdio_bus: miibus1: probed
stm32-usbphyc 5a006000.usbphyc@5a006000.of: registered rev:1.0
dwc2 49000000.usb-otg@49000000.of: EPs: 9, dedicated fifos, 0x3b8 entries in SPRAM
stm32_sdmmc 58007000.mmc@58007000.of: registered as mmc1
mmc1: detected MMC card version 5.1
mmc1: registered mmc1.boot0
mmc1: registered mmc1.boot1
mmc1: registered mmc1
stm32-iwdg 5a002000.watchdog@5a002000.of: probed
state: New state registered 'state'
state: Using bucket 0@0x00000000
malloc space: 0xcfdfde20 -> 0xdfbfbc3f (size 254 MiB)
eth0: got preset MAC address: 18:74:e2:a0:00:c0
eth1: got preset MAC address: 18:74:e2:a0:00:c1
eth2: got preset MAC address: 18:74:e2:a0:00:c2
envfs: no envfs (magic mismatch) - envfs never written?
multi_bind: creating Fastboot function
multi_bind: creating ACM function
g_multi gadget0: Multifunction Composite Gadget
g_multi gadget0: g_multi ready
dwc2 49000000.usb-otg@49000000.of: bound driver g_multi

Hit m for menu or any to stop autoboot:    1
barebox@Linux Automation Test Automation Controller (TAC) Gen 2:/ dhcp eth1
T eth1: DHCP client bound to address 192.168.0.240
barebox@Linux Automation Test Automation Controller (TAC) Gen 2:/ ping 192.168.0.1
PING 192.168.0.1 (192.168.0.1)
host 192.168.0.1 is alive
barebox@Linux Automation Test Automation Controller (TAC) Gen 2:/ boot
Booting entry 'bootchooser'
Booting entry 'root0'
fb0: framebuffer console 30x15 activated
…
```

hnez commented:
Assigned @ukleinek because he also did some work on the LXA TAC barebox patch stack, but we had a bit of mis-communication and now both have our own versions of the patch stack.
I'd like to leave the decision on how continue with him.

ukleinek commented on meta-lxatac-bsp/recipes-bsp/barebox/files/lxatac/defconfig:
```diff
@@ -52,11 +49,11 @@ CONFIG_CPU_32v7=y
-CONFIG_MACH_STM32MP15XX_DKX=y
-CONFIG_MACH_LXA_MC1=y
+# CONFIG_MACH_STM32MP15XX_DKX is not set
+# CONFIG_MACH_LXA_MC1 is not set
```
That seems unrelated to a pure bump

hnez replied:
Did you notice that this change is in a separate commit from the version bump?

ukleinek commented on meta-lxatac-bsp/recipes-bsp/barebox/files/lxatac/defconfig:
```diff
@@ -52,11 +49,11 @@ CONFIG_CPU_32v7=y
-CONFIG_MACH_SEEED_ODYSSEY=y
-CONFIG_MACH_STM32MP15X_EV1=y
+# CONFIG_MACH_SEEED_ODYSSEY is not set
+# CONFIG_MACH_STM32MP15X_EV1 is not set
```
ditto

ukleinek commented on meta-lxatac-bsp/recipes-bsp/barebox/files/lxatac/defconfig:
```diff
@@ -485,10 +488,7 @@ CONFIG_OF_BAREBOX_DRIVERS=y
 CONFIG_OF_OVERLAY=y
 # CONFIG_OF_OVERLAY_LIVE is not set
-CONFIG_AIODEV=y
-# CONFIG_LM75 is not set
-# CONFIG_ST_GYRO is not set
-CONFIG_STM32_ADC=y
+# CONFIG_AIODEV is not set
```
similar change unrelated to a version bump

ukleinek replied:
Oh, it seems I missed the context of this change. In this "disable features we do not need" commit the change looks fine.

ukleinek commented on meta-lxatac-bsp/recipes-bsp/barebox/files/lxatac/defconfig:
```diff
@@ -622,7 +625,7 @@ CONFIG_USB_GADGET_FASTBOOT=y
 CONFIG_VIDEO=y
 CONFIG_FRAMEBUFFER_CONSOLE=y
 # CONFIG_DRIVER_VIDEO_FB_SSD1307 is not set
-# CONFIG_VIDEO_VPL is not set
+CONFIG_VIDEO_VPL=y
```
ditto, and some more further down

ukleinek commented on meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0011-scripts-add-bareboxtlv-host-target-tool.patch:
```diff
@@ -225,8 +225,7 @@ index 000000000000..eca430104af9
 +		}
-+		if (ret >= 0)
-+			ret = write_full_verbose(1, p, size);
++		ret = write_full_verbose(1, p, size);
```
This change is fine (because ret is never negative here), but I'd mention that in the commit log, at least with something like "simplification in bareboxtlv host tool".

ukleinek commented on meta-lxatac-bsp/recipes-bsp/barebox/files/patches/0102-video-Add-of_get_display_timing.patch:
```diff
@@ -1,61 +0,0 @@
-From: Philipp Zabel <p.zabel@pengutronix.de>
-Date: Mon, 27 Mar 2023 14:17:09 +0200
-Subject: [PATCH] video: Add of_get_display_timing
-
-Add a port of the kernel of_get_display_timing() that writes a
-struct fb_videomode instead of struct display_timing, which we
-don't have.
-
-Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
-Reviewed-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
----
- drivers/video/of_display_timing.c | 22 ++++++++++++++++++++++
- include/fb.h                      |  2 ++
- 2 files changed, 24 insertions(+)
-
```
I'd mention this change as:
patch 0102-video-Add-of_get_display_timing was included in v2023.05.0 as commit 9190f984ec2c

Similar for the other dropped patches.

hnez replied:
Would you be okay with me not digging through old patches to add these annotations now?

ukleinek commented on meta-lxatac-bsp/recipes-bsp/barebox/files/patches/series.inc:
```diff
@@ -1,57 +1,32 @@
-  file://patches/0010-common-add-barebox-TLV-support.patch \
-  file://patches/0011-commands-add-TLV-debugging-command.patch \
-  file://patches/0012-scripts-add-bareboxtlv-host-target-tool.patch \
+  file://patches/0005-common-add-optional-systemd.hostname-generation.patch \
+  file://patches/0006-net-factor-out-eth_of_get_fixup_node.patch \
+  file://patches/0007-net-export-list-of-registered-ethernet-addresses.patch \
+  file://patches/0008-net-implement-ethaddr_string_cmp.patch \
+  file://patches/0009-common-add-barebox-TLV-support.patch \
+  file://patches/0010-commands-add-TLV-debugging-command.patch \
+  file://patches/0011-scripts-add-bareboxtlv-host-target-tool.patch \
```
If `umpf format-patch -n` was used the diff would be smaller and so easier to review.
(But I agree this is a subjective preference.)

ukleinek commented on meta-lxatac-bsp/recipes-core/images/emmc-boot-image/lxatac/genimage.config:
```diff
@@ -18,7 +18,7 @@ image @IMAGE@fip {
         fw-config = "stm32mp157c-lxa-tac-fw-config.dtb"
         hw-config = "stm32mp157c-lxa-tac-fw-config.dtb"
-        nt-fw = "barebox-stm32mp15xc-lxa-tac.img"
+        nt-fw = "barebox-stm32mp15xc-lxa-tac.stm32.img"
```
This also looks like a change that is worth explaining in the commit log.

hnez committed:
meta-lxatac-bsp: barebox: update to config that is actually used

This is just a menuconfig run without changing any options.
The VIDEO_VPL config option is automatically selected by the following
combination of other options:

  DRIVER_VIDEO_PANEL_MIPI_DBI [=y] && VIDEO [=y] && OFTREE [=y] && SPI [=y]

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-bsp: barebox: disable boards we do not actually support

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-bsp: barebox: disable features we do not need

It is neat to have some debugging features in the bootloader
when you need them.
But we have accumulated a few features that we definitely do
not use.

Disable those.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>

hnez committed:
meta-lxatac-bsp: barebox: enable fastboot oem command

This allows us to run barebox commands without the need for a serial
console (using the dedicated serial cable or the barebox CDC ACM gadget).

This solves an issue where the TAC would just continue auto booting
when trying to flash a new image onto the TAC via USB.

Also document in the README that we no longer need workarounds for
the watchdog and can now boot the flashed image without requiring a
power cycle.

Signed-off-by: Leonard Göhrs <l.goehrs@pengutronix.de>
